lib/rules/v-if-else-key.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * @author Felipe Melendez
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// =============================================================================
+// Requirements
+// =============================================================================
+
+const utils = require('../utils')
+const casing = require('../utils/casing')
+
+// =============================================================================
+// Constants and Variables
+// =============================================================================
+
+/**
+ * Map to store conditionally rendered components and their respective conditional directives.
+ *
+ * A conditional family is a group of components that are conditionally rendered using v-if, v-else-if, and v-else.
+ * This data structure helps track and manage the relation of such components with one another.
+ *
+ * This map links a parent node to its associated conditional family, representing
+ * the relationship between different parts of a Vue conditional rendering chain
+ * (v-if, v-else-if, v-else). Each entry in the map associates a parent node
+ * (representing the scope of the conditional family) with an object that tracks
+ * the nodes for each part of the conditional chain:
+ *
+ *  - 'if': Represents the node associated with the 'v-if' directive.
+ *  - 'elseIf': An array representing nodes associated with any 'v-else-if' directives.
+ *  - 'else': Represents the node associated with the 'v-else' directive.
+ *
+ * This structure helps in understanding the relationship between conditional directives
+ * within a given scope and aids in linting scenarios where these relationships matter.
+ *
+ * @type {Map<VElement, { if: VElement | null, elseIf: VElement[], else: VElement | null }>}
```

Comment:
Since you are reusing this type, you can define it once using `@typedef` and then use its name everywhere:

```js
/**
 * @typedef ConditionalFamily
 * @property {VElement | null} if
 * @property {VElement[]} elseIf
 * @property {VElement | null} else
 */

/** @type {Map<VElement, ConditionalFamily>} */

By the way, does the `if` property have to be nullable or can we make it required?

Reply:
defined the type, refactored to made `if` required, and fine-tuned the docstrings, they were too verbose: [1206bbd](https://github.com/vuejs/eslint-plugin-vue/pull/2280/commits/1206bbd1a8b1238c18a129c1e6d1228baa7d1067)